app/src/main/java/com/nutomic/syncthingandroid/activities/SettingsActivity.java

```diff
@@ -261,6 +265,7 @@ public void onApiChange(SyncthingService.State currentState) {
             mRelaysEnabled.setChecked(mOptions.relaysEnabled);
             mGlobalAnnounceServers.setText(joiner.join(mOptions.globalAnnounceServers));
             mAddress.setText(mGui.address);
+            mRestartOnWakeup.setChecked(mOptions.restartOnWakeup);
```

Comment:
I don't see this being user anywhere where it would actually set the setting?

Reply:
useFolderObserver is toggled by the user and does only affect the wrapper. Therefore we have the ChangeListener.
mRestartOnWakeup is derived from syncthing core, see Options.java (if I remember it correct), so the app just reflects what is set within syncthing core. If changed in the wrapper's UI, it is written back to syncthing core and the core is restarted.